tutorial/consuming_packages/cross_building_with_conan.rst

```diff
@@ -0,0 +1,217 @@
+    build_type=Release
+    compiler=gcc
+    compiler.cppstd=gnu14
+    compiler.libcxx=libstdc++11
+    compiler.version=9
+
+And the profile for the Raspberry Pi that is the **host** machine could look like this:
+
+.. code-block:: bash
+    :caption: <conan home>/profiles/raspberry
+    :emphasize-lines: 9-12
+
+    [settings]
+    os=Linux
+    arch=armv7hf
+    compiler=gcc
+    build_type=Release
+    compiler.cppstd=gnu14
+    compiler.libcxx=libstdc++11
+    compiler.version=9
+    [buildenv]
+    CC=arm-linux-gnueabihf-gcc-9
+    CXX=arm-linux-gnueabihf-g++-9
+    LD=arm-linux-gnueabihf-ld
+
+.. important::
+
+    Please, take into account that in order to build this example successfully, you should
+    have installed a toolchain that includes the compiler and all the tools to build the
+    application for the proper architecture. In this case the host machine is a Raspberry
+    Pi 3 with *armv7hf* architecture operating system and we have the
+    *arm-linux-gnueabihf* toolchain installed in the Ubuntu machine.
+
+If you have a look at the *raspberry* profile, you will see a section named
+``[buildenv]``. This section is used to set the environment variables that are needed to
+build the application. In this case we declare the ``CC``, ``CXX`` and ``LD`` variables
+pointing to the cross-build toolchain compilers and linker, respectively. This information
+will be later set in the ``conanbuild.sh`` script that we will source before building with
+CMake so that it can use the cross-build toolchain.
+
+Build and host contexts
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we have our two profiles prepared, let's have a look at our *conanfile.py*:
+
+.. code-block:: python
+    :caption: **conanfile.py**
+
+    from conan import ConanFile
+    from conan.tools.cmake import cmake_layout
+
+    class CompressorRecipe(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeToolchain", "CMakeDeps"
+
+        def requirements(self):
+            self.requires("zlib/1.2.11")
+            self.tool_requires("cmake/3.19.8")
+
+        def layout(self):
+            cmake_layout(self)
+
+As you can see, this is practically the same *conanfile.py* we used in the previous
+example. We will require **zlib/1.2.11** as a regular dependency and **cmake/3.19.8** as a
+tool needed for building the application. Also, we are using the pre-defined
+``cmake_layout``.
+
+We will need the application to build for the Raspberry Pi with the cross-build
+toolchain and also link the **zlib/1.2.11** library built for the same platform. On the
+other side, we need the **cmake/3.19.8** binary to run in Ubuntu Linux. Conan manages this
+internally in the dependency graph differentiating between what we call the "build
+context" and the "host context":
+
+* The **host context** is populated with the root package (the one specified in the
+  :command:`conan install` or :command:`conan create` command) and all its requirements.
```

Comment:
mention `self.requires()` ?

Reply:
I think having it in the recipe is enough and adding it here will probably make the explanation more complex to understand

Reply:
My point is, `and all its requirements.`  might be misleading, the `CMake` package is also a requirement.

Reply:
Ok, I'll clarify that those are added via self.requires()

Reply:
In that case, to refer to CMake and so, it could be `its tool-requirements` then 😆 but I understand the point that it could be misleading. Up to @czoido choice in my case 🤓